### PR TITLE
[MBL-1738] iPadOS 18 - Navbar overlay issue in "Follow Friends" screen

### PR DESCRIPTION
**Pull request by jovaniks**
# 📲 What

- Moved the call to `setNavigationBarHidden` from `viewWillAppear` to `viewDidAppear` in the "Follow Friends" screen.

# 🤔 Why

The change was needed to fix an issue where the `UINavigationBar` was overlaying the content of the "Follow Friends" screen when running the app on a device with iPadOS 18 and Xcode 16.

This issue appears to be caused by layout changes introduced in iPadOS 18, where the tab bar is now at the top of the screen. By calling `setNavigationBarHidden` in `viewDidAppear` instead of `viewWillAppear`, we ensure that the layout has been fully constructed, which prevents the `UINavigationBar` from incorrectly overlaying the content.

# 🛠 How

The `setNavigationBarHidden` function was moved from `viewWillAppear` to `viewDidAppear` so that the layout is fully built before manipulating the visibility of the navigation bar. This change resolves the layout issue observed in iPadOS 18.

# 👀 See

[Jira issue](https://kickstarter.atlassian.net/browse/MBL-1738)

| Before 🐛 | After 🦋 |
| --- | --- |
| image | image |

# ✅ Acceptance criteria

- [ ] Ensure the `UINavigationBar` no longer overlays the "Follow Friends" screen on iPadOS 18.

### CODE REVIEW

**Review by ifosli — COMMENTED** [review]
Since this is building on your other prs, please either a) wait for them to be submitted before marking this as ready for review or b) create a new branch that only contains the required changes (ie it only moves the `setNavigationBarHidden` call).

**Comment by nativeksr** [comment]
### SwiftFormat found issues:

| File | Rules |
| ---- | ----- |
| Kickstarter-iOS/Features/FindFriends/Controller/FindFriendsViewController.swift:38:1 | warning: (trailingSpace) Remove trailing space at end of a line. |


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Kickstarter-iOS/Features/FindFriends/Controller/FindFriendsViewController.swift [hunk 1]
```diff
@@ -33,8 +33,8 @@ internal final class FindFriendsViewController: UITableViewController {
     self.viewModel.inputs.viewDidLoad()
   }
 
-  override func viewWillAppear(_ animated: Bool) {
-    super.viewWillAppear(animated)
+  override func viewDidAppear(_ animated: Bool) {
+    super.viewDidAppear(animated)
```
Comment (stevestreza-ksr) [inline Kickstarter-iOS/Features/FindFriends/Controller/FindFriendsViewController.swift:37]: Is `animated` ever true here? By changing this from `viewWillAppear` to `viewDidAppear`, if animated is true, the nav bar will animate away after the view is on screen.

Comment (jovaniks) [reply Kickstarter-iOS/Features/FindFriends/Controller/FindFriendsViewController.swift:37]: yes, the `animated` is `true`, so the nav bar is appear after the view is on screen, I attached a video:

https://github.com/user-attachments/assets/c628b925-83c6-4c09-aea2-9e175b4a0361

**Review by scottkicks — COMMENTED** [review]
LGTM but let's clean up the commit history as ingerid suggests before merging.

**Pull request closed without merging**